packages/sakuli-core/src/bootstrap/get-node-modules-paths.function.ts

```diff
@@ -0,0 +1,26 @@
+import { join, resolve } from "path";
+import execa from "execa";
+import { isExistingDirectory } from "./is-existing-directory.function";
+
+export const getNodeModulesPaths = async (
+  path: string = "."
+): Promise<string[]> => {
+  let nodeModulesPaths: string[] = [];
+  let previous: string;
+
+  do {
+    const modules = join(path, "node_modules/@sakuli");
+    if (isExistingDirectory(modules)) {
+      nodeModulesPaths.push(modules);
+    }
+
+    previous = path;
+    path = resolve(join(previous, ".."));
+  } while (previous !== path);
```

Comment:
Do I understand ist correctly that you scan every directory from `${workdir}` to `/` for `node_modules/@sakuli` subfolders? Why is this required? Wouldn't it be enough to scan the Sakuli project structure? We should not explore directories on a customers machine that don't belong to sakuli test suites.

In addition, the `@sakuli` search criteria is a strong restriction to presets developed by customers/users. They would have to publish their packages under `@sakuli` to leverage auto detection. They could still register it manually in `package.json` so it is not breaking the possibility to register custom presets but this is something to keep in mind.

Reply:
Good point. I changed it so that only the global and the package level node_modules are searched.

I have limited this to `@sakuli` because there will be projects that could have hundreds of modules. So all modules would have to be searched for the `sakulipreset` keyword.

Reply:
A absolutely agree with the `@sakuli` limitation on automated search. Searching for presets in all `node_module` packages is the proverbial search for a needle in a haystack. It is just something, we have to keep in mind. In addition, I recognized manual loading was removed. Maybe we should keep it as a fallback solution for customer packages?